script/tool/lib/src/native_test_command.dart

```diff
@@ -0,0 +1,377 @@
+  }
+
+  Future<_PlatformResult> _testAndroid(Directory plugin, _TestMode mode) async {
+    final List<Directory> examplesWithTests = <Directory>[];
+    for (final Directory example in getExamplesForPlugin(plugin)) {
+      if (!isFlutterPackage(example)) {
+        continue;
+      }
+      if (example
+              .childDirectory('android')
+              .childDirectory('app')
+              .childDirectory('src')
+              .childDirectory('test')
+              .existsSync() ||
+          example.parent
+              .childDirectory('android')
+              .childDirectory('src')
+              .childDirectory('test')
+              .existsSync()) {
+        examplesWithTests.add(example);
+      } else {
+        _printNoExampleTestsMessage(example, 'Android');
+      }
+    }
+
+    if (examplesWithTests.isEmpty) {
+      return _PlatformResult(RunState.skipped);
+    }
+
+    bool failed = false;
+    bool hasMissingBuild = false;
+    for (final Directory example in examplesWithTests) {
+      final String exampleName = getPackageDescription(example);
+      _printRunningExampleTestsMessage(example, 'Android');
+
+      final Directory androidDirectory = example.childDirectory('android');
+      final File gradleFile = androidDirectory.childFile(_gradleWrapper);
+      if (!gradleFile.existsSync()) {
+        printError('ERROR: Run "flutter build apk" on $exampleName, or run '
+            'this tool\'s "build-examples --apk" command, '
+            'before executing tests.');
+        failed = true;
+        hasMissingBuild = true;
+        continue;
+      }
+
+      final int exitCode = await processRunner.runAndStream(
+          gradleFile.path, <String>['testDebugUnitTest', '--info'],
+          workingDir: androidDirectory);
+      if (exitCode != 0) {
+        printError('$exampleName tests failed.');
+        failed = true;
+      }
+    }
+    return _PlatformResult(failed ? RunState.failed : RunState.succeeded,
+        error:
+            hasMissingBuild ? 'Examples must be built before testing.' : null);
+  }
+
+  Future<_PlatformResult> _testIos(Directory plugin, _TestMode mode) {
+    return _runXcodeTests(plugin, 'iOS', mode,
+        extraFlags: _iosDestinationFlags);
+  }
+
+  Future<_PlatformResult> _testMacOs(Directory plugin, _TestMode mode) {
+    return _runXcodeTests(plugin, 'macOS', mode);
+  }
+
+  /// Runs all applicable tests for [plugin], printing status and returning
+  /// the test result.
+  ///
+  /// The tests targets must be added to the Xcode project of the example app,
+  /// usually at "example/{ios,macos}/Runner.xcworkspace".
+  Future<_PlatformResult> _runXcodeTests(
+    Directory plugin,
+    String platform,
+    _TestMode mode, {
+    List<String> extraFlags = const <String>[],
+  }) async {
+    String? testTarget;
+    if (mode.unitOnly) {
+      testTarget = 'RunnerTests';
+    } else if (mode.integrationOnly) {
+      testTarget = 'RunnerUITests';
```

Comment:
You don't need to address it in this PR since I know this was copied, but I just remembered about the image_picker `RunnerUITestiOS14` target.  It probably doesn't make sense for that to be a separate target, I wasn't involved in the recent creation of that.

Reply:
Yes, I found that when checking the repo to make sure this would work, and discussed it with folks this morning; it should get folded back in in the next day or two.